storages/backends/gcloud.py

```diff
@@ -319,17 +327,14 @@ def url(self, name, parameters=None):
                 quoted_name=_quote(name, safe=b"/~"),
             )
         else:
-            default_params = {
-                "bucket_bound_hostname": self.custom_endpoint,
+            params = {
+                "service_account_email": self.credentials.service_account_email,
+                "access_token": self.credentials.token,
```

Comment:
currently self.credentials throws a attribute error here as well since self.credentials hasn't been changed. the credentials that were set in the client @property doesn't flow over here. Might have to either set the self.credentials in there, or use self._client._credentials. 1st one seems ugly in setting a instance variable in an other property, 2nd option uses private instance variable. There is probably a 3rd option but that might involve some additional clean up